### ISSUE TITLE
Processes not ending properly

### ISSUE THREAD

**Issue by Frijol** [tessel/t2-cli#229]
CLI isn't telling Node to end the process at end of script or CTRL+C, so multiple processes stay open, resulting in ECONNREFUSED eventually

**Comment by flaki** [comment]
I can reproduce these. Also, module port connections remain open in the running process so the next `t2 run` that uses module ports fails. One has to manually `killall node` from ssh/console to be able to execute code again.

**Comment by johnnyman727** [comment]
Can also reproduce. I investigated very shortly and found that the CLI doesn't seem to be killing the process but it does die. Somehow it's not giving up the socket? Will need to investigate further.

**Comment by johnnyman727** [comment]
Doing some more investigating.

It looks like I can run the default `t2 init` blinky script and cancel it with control+c just fine. This may only be an issue with scripts that go to completion.

**Comment by johnnyman727** [comment]
I can also confirm that this appears to only be an issue when `require('tessel');` is involved.

**Comment by johnnyman727** [comment]
It seems like there are two separate issues at play here:

1. The process isn't closed once a script 'completes', where 'completes' means there is nothing left in the event queue. The cause of this issue is `net.createConnection` as it's used to open domain sockets to the spi daemon. This is necessary to communicate with the samd21. The connection could have data coming in at any time so the script decides it shouldn't close. I'm not sure how to get around this - @rwaldron any thoughts?

2. You process doesn't get closed when sent a SIGINT signal. This one is more of a mystery to me. Once the script starts, there are actually two running process:
```
 2308 root      1484 S    ash -c node '/tmp/remote-script/index.js'
 2309 root     43572 S    node /tmp/remote-script/index.js
```
You can manually `kill -9` `2308` and `2309` will still live but if you kill `2309`, both are killed. If you manually run `ash -c node 'tmp/remote-script/index.js'` in the console, you get a Node REPL...

I've also [tried forwarding the SIGINT signal to the remote process](https://github.com/tessel/t2-cli/pull/260) (which probably should have been in the CLI in the first place) but it doesn't seem to kill the processes. My guess is that the SIGINT is actually being directed at the `ash` process which doesn't die by a SIGINT (I've tested this on the console). Perhaps it's the equivalent of control+c'ing in a REPL and getting the `(^C again to quit)` message? I'm also not really sure how to tackle this one. We could run `killall node` afterward but that seems pretty dirty...

**Comment by johnnyman727** [comment]
@kevinmehall you might have some suggestions on the previous comment as well. I'm sort of at a loss at the moment.

**Comment by kevinmehall** [comment]
For the first issue, use https://nodejs.org/api/net.html#net_socket_unref. It will need to keep track of when you are waiting on IO and re-`ref` the socket so the process doesn't exit, and then `unref` again when there is no IO pending.

**Comment by kevinmehall** [comment]
For the second issue, try adding a call to http://linux.die.net/man/2/setsid or http://linux.die.net/man/2/setpgid after forking in usbexecd. See https://en.wikipedia.org/wiki/Process_group.

Also test that if a script spawns child processes that they get killed correctly, as that's a similar issue.

**Comment by johnnyman727** [comment]
Fixed with #94 and #93.
